src/common/pick_address.cc

```diff
@@ -112,6 +112,13 @@ void pick_addresses(CephContext *cct, int needs)
     exit(1);
   }
 
+  if (!cct->_conf->public_network.empty()
+       && cct->_conf->cluster_network.empty()) {
+    lderr(cct) << "Public network was set, but cluster network was not set " << dendl;
+    lderr(cct) << "    Using public network also for cluster network" << dendl;
+    cct->_conf->set_val("cluster_network", cct->_conf->public_network);
+  }
```

Comment:
@liewegas 
I think that could be done even simpler, and just pass public_network instead of cluster_network in that case.
I'll rewrite the patch.